.github/workflows/publish-py.yml

```diff
@@ -16,14 +16,17 @@ jobs:
               uses: actions/setup-python@v5
               with:
                   python-version: "3.x"
-            - name: Install dependencies
+                  cache: pip
+            - name: Install dependecies
               run: |
-                  python -m pip install --upgrade pip 
-                  pip install -r requirements/build-src.txt
-            - name: Build and publish
+                  pip install --upgrade hatch uv
+            - name: Build
+              run: |
+                  hatch build --clean
+            # TODO: Update trusted publishing with pypa/gh-action-pypi-publish
```

Comment:
I don't mind making the switch to `pypa/gh-action-pypi-publish` within this PR, since we're already touching this file anyways

Reply:
Yes, it would take some work on PyPi (https://docs.pypi.org/trusted-publishers/).  

I was thinking to do that in a separate PR, this on was growing enough already!  Perhaps doing some updates to create a GitHub release with artifacts, publish, etc, on the creation of a semver tag?  Since this uses Keep a Changelog, even that can be pulled out automatically and added to the release

Reply:
As an idea of how I suggest updating the actions, I use this flow: https://github.com/stumpylog/gotenberg-client/actions/runs/10653499633

Basically, the checks like tests, documentation and building a wheel/sdist always run, then a release to Github and Pypi run when a tag is created to do those only on a tag.  This would also include publishing the versioned documentation.

If that sounds alright, I'm happy to work on it (in a separate PR)

Reply:
Unless I'm misunderstanding your comment, that is effectively what is currently being done in this repo but in an [event driven](https://github.com/Archmonger/ServeStatic/blob/3e63ed0b1b07d66ed1c9527d8c7a2eb93ad8483b/.github/workflows/publish-release-docs.yml#L3-L5) fashion rather than polling on each merge.

Reply:
Ahhh okay I understand your point - Yeah manually creating the GH releases is unnecessary manual labor. If we can automate that would be fantastic.